dtfs-central-api/src/services/portal/facility.service.ts

```diff
@@ -0,0 +1,39 @@
+import { AuditDetails, DEAL_TYPE, DealType, FacilityStatus } from '@ukef/dtfs2-common';
+import { ObjectId } from 'mongodb';
+import { updateFacilityStatus } from '../../v1/controllers/portal/facility/update-facility-status.controller';
+import { updateFacility } from '../../v1/controllers/portal/gef-facility/update-facility.controller';
+
+export class PortalFacilityService {
+  /**
+   * Updates the deal status
+   *
+   * @param updateStatusParams
+   * @param updateStatusParams.dealId - the deal Id to update
+   * @param updateStatusParams.newStatus - the status change to make
+   * @param updateStatusParams.auditDetails - the users audit details
+   * @param updateStatusParams.dealType - the deal type
+   */
+  public static async updateStatus({
+    facilityId,
+    status,
+    auditDetails,
+    dealType,
+  }: {
+    facilityId: ObjectId | string;
+    status: FacilityStatus;
+    auditDetails: AuditDetails;
+    dealType: DealType;
+  }): Promise<void> {
+    if (dealType === DEAL_TYPE.GEF) {
+      await updateFacility({ facilityId, facilityUpdate: { status }, auditDetails });
```

Comment:
`status` field seems good to me. I didn't realise GEF facilities had no `status` field 😮

Reply:
I'm considering changing this to `stage` since thats how its rendered in `gef-ui` (screenshot attatched). 
The pros of this
- makes whats rendered to the user consistent with whats stored in the database
The cons
- inconsistent with the `status` in BSS/EWCS facilities (though these two objects are completely different already)


![image](https://github.com/user-attachments/assets/aff065a5-d2e2-4d8d-b5a4-3687200ac3c7)

Reply:
Think i'll just stick with `status`